### PR TITLE
Various fixes in STM32F0

### PR DESCRIPTION
**Pull request by Delsian**
Various fixes according to comment #1422 (comment)
* Copyright in log.c
* GPIO check disabled
* generic/timer_irq added
* Change in doc section
PS: Updated variant of #1449

### CODE REVIEW

**Comment by Delsian** [comment]
Just FYI: timer_dispatch_many() has different implementations in timer_irq.c (returns abs time) and armcm_timer.c (returns diff time)

This may confuse somebody (like me), maybe have sense to use different names for both functions?

**Comment by KevinOConnor** [comment]
Thanks.  Some comments:
- the commit message needs to have a signed-off-by line (as described at https://github.com/KevinOConnor/klipper/blob/master/docs/CONTRIBUTING.md )
- normally the timer code would use the board specific irqs instead of the systick timer.  (Using the board specific timer allows the code to use absolute timestamps and is thus a little more accurate - the reason armcm_timer.c doesn't use timer_irq.c is specifically because of the need to handle the oddity of using systick.)
- I think it would be preferable to avoid #ifdefs in the code.  Typically an option would be defined by adding it to the Kconfig file (it does not need to be a user visible option), and then any C functions that are not needed when the option is disabled would start with `if (!CONFIG_MY_OPTION) return;`.

-Kevin

**Comment by Delsian** [comment]
Fixed.
And GPIO check was removed as it discussed in https://github.com/KevinOConnor/klipper/issues/1436#issuecomment-477318535

File: src/stm32f0/i2c.c [hunk 1]
```diff
@@ -50,26 +48,37 @@ struct i2c_config i2c_setup(uint32_t bus, uint32_t rate, uint8_t addr)
...
 {
+    uint32_t timeout = timer_read_time() + timer_from_us(5000);
     i2c_write(config, reg_len, reg);
     while(HAL_I2C_Master_Receive(&hi2c1, config.addr, read,
             (uint16_t)read_len, (uint32_t)1000)!= HAL_OK) {
          if (HAL_I2C_GetError(&hi2c1) != HAL_I2C_ERROR_AF) {
              shutdown("Buffer error");
          }
      }
+    i2c_to_check(timeout);
```
Comment (KevinOConnor) [inline src/stm32f0/i2c.c]: Typo?  Check should be in the loop?

-Kevin

Comment (Delsian) [reply src/stm32f0/i2c.c]: Ouch... :)

Comment (Delsian) [reply src/stm32f0/i2c.c]: Sure!

**Pull request merged**